### PR TITLE
files to test bug

### LINKED ISSUE #121: Possible bug with identifying free variable
I was using a very basic file to test model functions I hit the apparently unreachable code which says I have no free variable in the model (line 520 model.py)

Stepping through the code that looks for the free variable loops through the nodes of the graph. However, when creating the graph, a variable (state/free) of a derivative is only added as a node if it occurs on the rhs of the derivative; which since model uses dsv11/dt = 1 't' does not appear on rhs. So the Model object marks it as a free variable BUT no graph node ever gets created so it cannot be looked up.

Is this a bug ?

### PR DESCRIPTION
**Pull request by skeating**
Fixes #121

### CODE REVIEW

**Comment by jonc125** [comment]
The 'fixes #121' should go in the issue description, not the title.

File: tests/test_model_functions.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+            {'units': 'mV', 'exponent': '1'}, {'prefix': 'micro', 'units': 'second', 'exponent': '-1'}],
+        'mM_per_ms': [{'units': 'mM'}, {'units': 'ms', 'exponent': '-1'}],
+        'ms_power_prefix': [{'prefix': '-3', 'units': 'second'}],
+        'ms_with_multiplier': [{'multiplier': 0.001, 'units': 'second'}],
+    })
+
+    @pytest.fixture
+    def model(scope='class'):
+        return cellmlmanip.load_model(
+            os.path.join(os.path.dirname(__file__), 'cellml_files', "basic_ode.cellml"))
```
Comment (MichaelClerx) [inline tests/test_model_functions.py]: Can we use `'` for both strings? :D

File: tests/test_model_functions.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+    @pytest.fixture
+    def other_model(scope='class'):
+        return cellmlmanip.load_model(
+            os.path.join(os.path.dirname(__file__), 'cellml_files',
+                         "aslanidi_model_2009.cellml"))
```
Comment (MichaelClerx) [inline tests/test_model_functions.py]: same

File: tests/test_model_functions.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+    # also tested in test_hodgkin
+    def test_get_state_symbols(self, model):
+        state_symbols = model.get_state_symbols()
+        assert len(state_symbols) == 1
```
Comment (MichaelClerx) [inline tests/test_model_functions.py]: Test that it's the correct one too?

File: cellmlmanip/model.py [hunk 4]
```diff
@@ -458,6 +458,12 @@ def graph(self):
                     graph.add_node(rhs, equation=sympy.Eq(rhs, dummy), variable_type='parameter')
                     graph.add_edge(rhs, lhs)
 
+            # check that the free variable if defined as a node
```
Comment (MichaelClerx) [inline cellmlmanip/model.py]: Could we move this up to around line 428?

Comment (skeating) [reply cellmlmanip/model.py]: Might get added twice - since it may occur on rhs of another equation and get added there

Comment (MichaelClerx) [reply cellmlmanip/model.py]: But you could do the same check as here, no?

Comment (MichaelClerx) [reply cellmlmanip/model.py]: oh ok get it now

**Pull request merged**